geode-core/src/main/java/org/apache/geode/internal/util/JavaWorkarounds.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.util;
+
+import java.util.Map;
+import java.util.function.Function;
+
+public class JavaWorkarounds {
+
+  // This is a workaround for computeIfAbsent not being concurrent in pre Java 9
```

Comment:
I'd maybe change the verbage here from "not being concurrent" to "unnecessarily taking out a write lock in the case where the entry exists already and only a read is necessary".  Maybe a bit more verbose but gets the point across.

Reply:
Any suggestions for the class name?